lib/connections/service.go

```diff
@@ -667,9 +665,8 @@ func (s *service) ListenerStatus() map[string]ListenerStatusEntry {
 
 		err := listener.Error()
 		if err != nil {
-			status.Error = err.Error()
-		} else {
-			status.OK = true
+			errStr := err.Error()
+			status.Error = &errStr
```

Comment:
Could you please explain the reason for the string pointer, I can't figure it out.

Reply:
To be able to produce `"error": null` in the JSON

Reply:
Hmm, yeah I see, should have guessed that...  
And grepping to see whether we already use something like this in other places pointed me at `events.Error` being used in pulling, for just that purpose: https://github.com/syncthing/syncthing/blob/master/lib/events/events.go#L485 (weird place for that function though, and orry for the bikeshedding).